.github/workflows/release-workflow.yml

```diff
@@ -0,0 +1,183 @@
+                  Install-Module -Name $RequiredModule -Repository:($($PSDependencies[$RequiredModule].Repository)) -RequiredVersion:($($PSDependencies[$RequiredModule].RequiredVersion)) -AllowPrerelease -Force
+              }
+          }
+
+  Manual-Approval-Release:
+    needs: [Check-PR-Labels]
+    environment: PublishToPSGallery
+    runs-on: ubuntu-latest
+    steps:
+      - name: Manual Approval for Release
+        run: echo "Awaiting approval from required reviewers before continuing"
+
+  Draft-GH-Release:
+    needs: [Manual-Approval-Release]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Build Draft Release
+        run: |
+          # Module Version
+          VERSION=$(grep -o '\d\+\.\d\+\.\d\+' /PowerShell/JumpCloud\ Module/JumpCloud.psd1)
+          TITLE="JumpCloud PowerShell Module v$VERSION"
+          # Get the changelog text between the latest release # and the next sequential "## SemanticVersionNumber"
+          CHANGELOG=$(cat /PowerShell/ModuleChangelog.md |awk "/^## $VERSION/{ f = 1; next } /## [0-9]+.[0-9]+.[0-9]+/{ f = 0 } f")
+          TAG="v$VERSION"
+          BODY="$TITLE $CHANGELOG"
+          # Post Draft Release
+          gh api \
+            --method POST \
+            -H "Accept: application/vnd.github+json" \
+            -H "X-GitHub-Api-Version: 2022-11-28" \
+            /repos/${{ github.repository_owner }}/${{ github.repository }}/releases \
+            -f tag_name=$TAG \
+          -f name=$TITLE \
+          -f body=$BODY \
+          -F draft=true \
+          -F generate_release_notes=false
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+  Deploy-Nupkg:
+    needs: Draft-GH-Release
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/cache@v3
+        with:
+          path: "/home/runner/.local/share/powershell/Modules/"
+          key: PS-Dependancies
+      - name: Build Nuspec
+        run: |
+          $ErrorActionPreference = 'Stop'
+          . "./PowerShell/Deploy/BuildNuspecFromPsd1.ps1" -RequiredModulesRepo PSGallery
+      - name: Add nuget sources
+        #env:
+          #AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          #AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        run: |
+          #Set-AWSCredential -AccessKey:($env:AWS_ACCESS_KEY_ID) -SecretKey:($env:AWS_SECRET_ACCESS_KEY)
+          #$CARepoEndpoint = "$(Get-CARepositoryEndpoint -Domain jumpcloud-artifacts -Region us-east-1 -Repository jumpcloud-nuget-modules -Format nuget)v3/index.json"
+          #dotnet nuget add source $CARepoEndpoint --name CodeArtifact --username aws --password (Get-CAAuthorizationToken -Domain:("jumpcloud-artifacts") -Region:("us-east-1")).AuthorizationToken
+          dotnet nuget add source "https://www.powershellgallery.com/api/v2/package" --name PSGallery
```

Comment:
Nuget is already installed so nice, no need to re-install 
https://github.com/actions/runner-images/blob/main/images/linux/Ubuntu2204-Readme.md